### PR TITLE
LPS-189953

### PR DESCRIPTION
**Pull request by ces-locpham**
Hi @tototrinh ,
Please help me review this ticket.
# Motivation
[LPS-189953 ](https://liferay.atlassian.net/browse/LPS-189953) [BE] Add checkbox configuration for Enabled Headless API
[LPS-189955 ](https://liferay.atlassian.net/browse/LPS-189955) [BE] Create API definition for GET request and [LPS-189955 ](https://liferay.atlassian.net/browse/LPS-189956) [BE] Generate image captcha for GET request

# Solution
LPS-189953
I add a configuration for enabling the headless API, if the configuration is not enabled, when user send the request, the server will send 404 response.
LPS-189955 and LPS-189956
I create GET Api for getting the captcha image.
TODO: Add a configuration for guest who can use it through the Captcha API.
cc @ces-quanhuynh  @tototrinh

### CODE REVIEW

File: modules/apps/captcha/captcha-rest-impl/rest-config.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiDir: "../captcha-rest-api/src/main/java"
+apiPackagePath: "com.liferay.captcha.rest"
+application:
+    baseURI: "/captcha-rest"
+    className: "CaptchaRestApplication"
+    name: "CaptchaRest"
```
Comment (tototrinh) [inline modules/apps/captcha/captcha-rest-impl/rest-config.yaml:6]: I think it should be like below that follow Liferay pattern
```
    className: "CaptchaRESTApplication"
    name: "Liferay.Captcha.REST"
```

File: modules/apps/captcha/captcha-rest-impl/rest-openapi.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+info:
+    description:
+        "CaptchaRest REST API. A Java client JAR is available for use with the group ID 'com.liferay', artifact ID
+        'com.liferay.captcha.rest.client', and version '1.0.0'."
+    license:
+        name: "Apache 2.0"
+        url: "http://www.apache.org/licenses/LICENSE-2.0.html"
+    title: "CaptchaRest"
```
Comment (tototrinh) [inline modules/apps/captcha/captcha-rest-impl/rest-openapi.yaml:8]: Same above, `Liferay Captcha REST` is better?

File: modules/apps/captcha/captcha-rest-impl/src/main/java/com/liferay/captcha/rest/internal/resource/v1_0/CaptchaResourceImpl.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	properties = "OSGI-INF/liferay/rest/v1_0/captcha.properties",
+	scope = ServiceScope.PROTOTYPE, service = CaptchaResource.class
+)
+public class CaptchaResourceImpl extends BaseCaptchaResourceImpl {
+
+	@Override
+	public Response getSimpleCaptcha() throws Exception {
+		CaptchaConfiguration captchaConfiguration =
+			_configurationProvider.getCompanyConfiguration(
+				CaptchaConfiguration.class, _contextCompany.getCompanyId());
```
Comment (tototrinh) [inline modules/apps/captcha/captcha-rest-impl/src/main/java/com/liferay/captcha/rest/internal/resource/v1_0/CaptchaResourceImpl.java:40]: CaptchaConfiguration has the scope as SYSEM, not COMPANY. Please double-check.

**Comment by ces-totrinh** [comment]
https://github.com/locpham97/liferay-portal/pull/46

**Pull request closed without merging**